examples-gallery/plot_parameter_estimation.py

```diff
@@ -0,0 +1,205 @@
+- :math:`u_3`: speed of the mass of the third system [m/s]
+- :math:`u_4`: speed of the mass of the fourth system [m/s]
+
+**Parameters**
+
+- :math:`m`: mass for both systems system [kg]
+- :math:`c`: damping coefficient for both systems [Ns/m]
+- :math:`k`: spring constant for both systems [N/m]
+- :math:`l_0`: natural length of the spring [m]
+
+"""
+# %%
+# Set up the equations of motion and integrate them to get the noisy measurements.
+#
+import sympy as sm
+import numpy as np
+import sympy.physics.mechanics as me
+import matplotlib.pyplot as plt
+from scipy.integrate import solve_ivp
+from opty import Problem
+from opty.utils import parse_free
+
+N = me.ReferenceFrame('N')
+O, P1, P2, P3, P4 = sm.symbols('O P1 P2 P3 P4', cls=me.Point)
+
+O.set_vel(N, 0)
+t = me.dynamicsymbols._t
+
+x1, x2, x3, x4 = me.dynamicsymbols('x1 x2 x3 x4')
+u1, u2, u3, u4 = me.dynamicsymbols('u1 u2 h3 u4')
+m, c, k, l0 = sm.symbols('m c k l0')
+
+P1.set_pos(O, x1 * N.x)
+P2.set_pos(O, x2 * N.x)
+P3.set_pos(O, x3 * N.x)
+P4.set_pos(O, x4 * N.x)
+P1.set_vel(N, u1 * N.x)
+P2.set_vel(N, u2 * N.x)
+P3.set_vel(N, u3 * N.x)
+P4.set_vel(N, u4 * N.x)
+
+body1 = me.Particle('body1', P1, m)
+body2 = me.Particle('body2', P2, m)
+body3 = me.Particle('body3', P3, m)
+body4 = me.Particle('body4', P4, m)
+bodies = [body1, body2, body3, body4]
+
+forces = [(P1, -k * (x1 - l0) * N.x - c * u1 * N.x),
+    (P2, -k * (x2 - l0) * N.x - c * u2 * N.x), (P3, -k * (x3 - l0) * N.x - c * u3 * N.x),
+    (P4, -k * (x4 - l0) * N.x - c * u4 * N.x)]
+
+kd = sm.Matrix([u1 - x1.diff(), u2 - x2.diff(), u3 - x3.diff(), u4 - x4.diff()])
+
+q_ind = [x1, x2, x3, x4]
+u_ind = [u1, u2, u3, u4]
+
+KM = me.KanesMethod(N, q_ind, u_ind, kd_eqs=kd)
+fr, frstar = KM.kanes_equations(bodies, forces)
+eom = kd.col_join(fr + frstar)
+sm.pprint(eom)
+
+rhs = KM.rhs()
+
+qL = q_ind + u_ind
+pL = [m, c, k, l0]
+
+rhs_lam = sm.lambdify(qL + pL, rhs)
+
+
+def gradient(t, x, args):
+    return rhs_lam(*x, *args).reshape(8)
+
+t0, tf = 0, 20
+num_nodes = 500
+times = np.linspace(t0, tf, num_nodes)
+t_span = (t0, tf)
+
+x0 = np.array([3, 3, 3, 3, 0, 0, 0, 0])
+pL_vals = [1.0, 0.25, 1.0, 1.0]
+
+resultat1 = solve_ivp(gradient, t_span, x0, t_eval = times, args=(pL_vals,))
+resultat = resultat1.y.T
+
+noisy = []
+np.random.seed(123)
+for i in range(4):
+    noisy.append(resultat[:, i] + np.random.randn(resultat.shape[0]) * 0.5 +
+        np.random.randn(1)*2)
+
+fig, ax = plt.subplots(figsize=(10, 5))
+for i in range(4):
+    ax.plot(times, noisy[i], label=f'measurement {i+1}', lw=0.5)
+plt.xlabel('Time')
+ax.set_title('Measurements of the position of the mass')
+ax.legend();
+
+plt.show()
+
+
+# %%
+# Set up the Estimation Problem.
+# --------------------------------
+#
+# If some measurement is considered more reliable, its weight w can be increased.
+#
+# objective = :math:`\int_{t_0}^{t_f} (w_1 (x_1 - noisy_{x_1})^2 + w_2 (x_2 - noisy_{x_2})^2 + w_3 (x_3 - noisy_{x_3})^2 + w_4 (x_4 - noisy_{x_4})^2)\, dt`
```

Comment:
Using $x_1 - x_1^m$ where "m" is for measured would make these equations simpler